ebms-payload/src/main/kotlin/no/nav/emottak/payload/juridisklogg/JuridiskLoggService.kt

```diff
@@ -3,58 +3,63 @@ package no.nav.emottak.payload.juridisklogg
 import io.ktor.client.HttpClient
 import io.ktor.client.call.body
 import io.ktor.client.engine.cio.CIO
+import io.ktor.client.plugins.contentnegotiation.ContentNegotiation
 import io.ktor.client.request.basicAuth
 import io.ktor.client.request.post
 import io.ktor.client.request.setBody
 import io.ktor.http.ContentType
 import io.ktor.http.contentType
+import io.ktor.serialization.kotlinx.json.json
 import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runBlocking
 import kotlinx.coroutines.withContext
 import kotlinx.serialization.Serializable
 import no.nav.emottak.message.model.Direction
 import no.nav.emottak.message.model.PayloadRequest
 import no.nav.emottak.payload.log
 import no.nav.emottak.util.getEnvVar
+import no.nav.emottak.util.marker
 
 class JuridiskLoggService() {
     private val juridiskLoggUrl = getEnvVar("APP_JURIDISKLOGG_URI", "https://app-q1.adeo.no/juridisklogg") + "/api/rest/logg"
     private val juridiskLoggStorageTime = getEnvVar("JURIDISKLOGG_STORAGE_TIME_YEARS", "1").toInt()
     private val userName = getEnvVar("JURIDESKLOGG_USERNAME", "dummyUsername")
     private val userPassword = getEnvVar("JURIDESKLOGG_PASSWORD", "dummyPassword")
 
-    init {
-        log.debug("Juridisk logg URL: $juridiskLoggUrl")
-        log.debug("Juridisk logg user: $userName")
-        log.debug("Juridisk logg password length: ${userPassword.length}")
-    }
-
     fun logge(payloadRequest: PayloadRequest) {
-        val httpClient = HttpClient(CIO)
+        val httpClient = HttpClient(CIO) {
+            install(ContentNegotiation) {
+                json()
+            }
+        }
         val request = JuridiskLoggRequest(
             payloadRequest.messageId,
             if (payloadRequest.direction == Direction.IN) "Ekstern bruker" else "NAV",
             if (payloadRequest.direction == Direction.IN) "NAV" else "Ekstern bruker",
             juridiskLoggStorageTime,
-            payloadRequest.payload.bytes
+            java.util.Base64.getEncoder().encodeToString(payloadRequest.payload.bytes)
         )
         log.debug("Juridisk logg forespørsel: $request")
 
-        val response = suspend {
+        runBlocking {
```

Comment:
Istedenfor runBlocking kan du bruke suspend.

Reply:
Opprettet en separat sak for det https://github.com/navikt/team-emottak-docs/issues/72